### PR TITLE
CMake: increase minimum required version to 3.1.0

### PR DESCRIPTION
**Pull request by tamiko**
Closes: https://github.com/dealii/dealii/issues/11615

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -33,17 +33,14 @@
...
 #
 # We support all policy changes up to version 3.1.0. Thus, explicitly set
 # all policies CMP0001 - CMP0054 to new for version 3.1 (and later) to
 # avoid some unnecessary warnings.
 #
-IF( "${CMAKE_VERSION}" VERSION_EQUAL "3.1" OR
-    "${CMAKE_VERSION}" VERSION_GREATER "3.1" )
-  CMAKE_POLICY(VERSION 3.1.0)
-ENDIF()
+CMAKE_POLICY(VERSION 3.1.0)
```
Comment (tamiko) [inline CMakeLists.txt:43]: I think we discussed raising to 3.1 in the near future back when we had to introduce this `CMAKE_POLICY` switch.

**Comment by tamiko** [comment]
I would like to get some sign-offs by my fellow developers on this one. Let's wait for some mor than the minimum required approvals.

For reference: 

CMake-3.1 was released early 2015. We currently require as minimum compiler version gcc-5.1 (released April 2015), or clang-4.0.0 (which was released July 2017).

**Review by kronbichler — APPROVED** [review]

**Comment by tjhei** [comment]
Is this one location the only problem for cmake 2.8?
I don't mind increasing the minimum to 3.1 if 2.8 gives us headaches and extra work. Most current OSes have moved to 3.x, I would think. Ubuntu 16.04 has cmake 3.5 for example.  Otherwise, it is easy enough to install cmake yourself, even on a cluster.

**Review by bangerth — APPROVED** [review]

**Review by bangerth — COMMENTED** [review]
Maybe also add a changelog entry?

**Review by Rombur — APPROVED** [review]

**Review by peterrum — APPROVED** [review]

**Review by drwells — APPROVED** [review]

**Comment by tamiko** [comment]
@bangerth Added a changelog entry!

@tjhei Yes, we could have written a simple workaround for the regex.

But I think it's overdue that we raise the version requirement :smile:

**Review by masterleinad — APPROVED** [review]

**Comment by tamiko** [comment]
6 approvals, I'd say that's a quorum of some sorts :-)

**Comment by masterleinad** [comment]
Would you mind squashing your commits a bit?

**Comment by tamiko** [comment]
@masterleinad Done.

**Pull request merged**